### ISSUE TITLE
🚧Create custom shader exceptions

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/Velaptor#450]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Create custom exceptions that can be thrown if something goes wrong with the shader compilation and linking process.
Currently, standard `Exception` types are being used but should be replaced with custom exceptions.

Contained in the `ShaderProgram` class, there are 3 instances of the base `Exception` being thrown.  These 3 should be replaced with custom exceptions.

### **Places To Replace With Custom Exceptions:**
- Shader Compiling (`ShadeProgram.Init()`):
  > **Note** 
  > The 2 exceptions below will use the new `ShaderCompileException`. Please use the same exception messages.
  - Location 1: https://github.com/KinsonDigital/Velaptor/blob/e8ff7c6ade6c873a8ee916166b5dd31904cd72a3/Velaptor/OpenGL/Shaders/ShaderProgram.cs#L176
  - Location 2: https://github.com/KinsonDigital/Velaptor/blob/e8ff7c6ade6c873a8ee916166b5dd31904cd72a3/Velaptor/OpenGL/Shaders/ShaderProgram.cs#L195
- Shader Linking (`ShaderProgram.CreateProgram()`)
  > **Note** 
  > The exception below will use the new `ShaderLinkException`, Please use the same exception messages.
  - Location 1: https://github.com/KinsonDigital/Velaptor/blob/88323346bb0bb9b5636fe09cb323ba91fc282275/Velaptor/OpenGL/Shaders/ShaderProgram.cs#L208

### **New exception names:**
- `ShaderCompileException`
- `ShaderLinkException`

> **Note**
> All custom exceptions go in the `Velaptor.OpenGL.Exceptions` namespace
> and the directory _**Velaptor/OpenGL/Exceptions**_.
> The unit tests for the exceptions will follow the same namespace and folder location pattern as the actual implementations
> All exceptions should be serializable, and `sealed`.  Use the exception `AtlasException` as a guide on the serializable implementation constructor.

### Acceptance Criteria

- [x] New exception with the name `ShaderCompileException` created
  - [x] Unit tests created
  - [x] Marked as `sealed`
  - [x] Uses the `Serializable` attribute
- [x] New exception with the name `ShaderLinkException` created
  - [x] Unit tests created
  - [x] Marked as `sealed`
  - [x] Uses the `Serializable` attribute
- [x] Standard `Exception`'s in the `Init()` method updated with the new `ShaderCompileException` type.
- [x] `CreateProgram()` method updated with new compile exception type.
- [x] Standard `Exception` in the `CreateProgram()` method updated with the new `ShaderLinkException` type
- [x] Unit tests updated
  > **Note**
  > These will be located in the _**VelaptorTests**_ project with the class name of `ShaderProgramTests`

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [x] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| Workflow Changes    | `workflow` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by softwareantics** [comment]
Happy to do this :)

**Comment by CalvinWilkinson** [comment]
@softwareantics I have moved this issue from release _**v1.0.0-preview.22**_ to _**v1.0.0-preview.23**_.

I am also going to be moving this project to a new branching model.  As of right now, do not start the work.  Please wait until we are finished with the conversion.  I cannot tell you how long it will exactly take but it won't be too long.

Sorry for the interruption but I needed to do a release so that way I can move forward with the new branching model changes.  So as of right now, hold tight.

Thanks!!
